modules/cannarithm/test/test_element_operation.cpp

```diff
@@ -0,0 +1,137 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "test_precomp.hpp"
+#include "opencv2/ts/cuda_test.hpp"
+#include "opencv2/cann_arithm.hpp"
+
+namespace opencv_test
+{
+namespace
+{
+// Random Generator
+Mat randomMat(int w, int h, int dtype)
+{
+    Mat rnMat(w, h, dtype);
+    RNG rng;
+    rng.fill(rnMat, RNG::UNIFORM, 0.f, 1.f);
+    return rnMat;
+}
+cv::Scalar randomScalar()
+{
+    RNG rng;
+    return Scalar(rng, rng.next(), rng.next(), rng.next());
+}
+float randomNum()
+{
+    RNG rng;
+    float rdnNum = float(rng.uniform(0.3, 3.0));
+    return rdnNum;
+}
+Mat genMask()
+{
+    Mat mask = Mat::zeros(Size(10, 10), CV_8UC1);
+    rectangle(mask, cv::Rect(5, 5, 3, 3), Scalar(255), -1);
+    return mask;
+}
+
+#define DEVICE_ID 0
+
+/****************TEST CASE***************/
+// MAT & Mat
+#define TEST_MAT_OP_MAT(idx, op, ...)                        \
+    TEST(ELEMENTWISE_OP, MAT_##op##_MAT_##idx)               \
+    {                                                        \
+        cv::cann::setDevice(DEVICE_ID);                      \
+                                                             \
+        Mat cpuMat1 = randomMat(10, 10, CV_32SC3);           \
+        Mat cpuMat2 = randomMat(10, 10, CV_32SC3);           \
+        Mat cpuDst;                                          \
+        cv::op(cpuMat1, cpuMat2, cpuDst, __VA_ARGS__);       \
```

Comment:
It is better to avoid multi-line macros.
Especially in tests, as we can't debug code line-by-line (macros is a single complex line of code)

Use templates instead.

Reply:
FIxed.  But it's still exists some multi-line macros. I will change them in next code refactoring(maybe in separate PR).